### PR TITLE
Add ForceRefresh to OBO request

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

**Comment by jennyf19** [comment]
@jmprieur @bgavrilMS use w/this [id web branch](https://github.com/AzureAD/microsoft-identity-web/pull/788)

File: src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenOnBehalfOfParameterBuilder.cs [hunk 1]
```diff
@@ -65,6 +65,26 @@ public AcquireTokenOnBehalfOfParameterBuilder WithSendX5C(bool withSendX5C)
             return this;
         }
 
+        /// <summary>
+        /// Specifies if the client application should force refreshing the
+        /// token from the user token cache. By default the token is taken from the
+        /// the application token cache (forceRefresh=false)
```
Comment (jmprieur) [inline src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenOnBehalfOfParameterBuilder.cs:71]: user token cache?

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenOnBehalfOfParameterBuilder.cs:71]: LGTM. @jennyf19.

File: src/client/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenOnBehalfOfParameters.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ internal class AcquireTokenOnBehalfOfParameters : IAcquireTokenParameters
     {
         public UserAssertion UserAssertion { get; set; }
         public bool SendX5C { get; set;}
+        public bool ForceRefresh { get; set; }
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenOnBehalfOfParameters.cs:13]: Please add a line in LogParamters about this (and maybe also log the UserAssertion.Hash)

**Comment by bgavrilMS** [comment]
LGTM so far, needs a unit test

**Comment by jennyf19** [comment]
yes. hence the draft. have added them now. i need a PR to see if the build would pass to run a release build to do the changes in id web.

---
In reply to: [735729925](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2250#issuecomment-735729925) [](ancestors = 735729925)

File: src/client/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenOnBehalfOfParameters.cs (hunk 2 above)
Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenOnBehalfOfParameters.cs:13]: done. 
not sure why to log the hash...is there a reason for that? i have not included it.

---
In reply to: [532528452](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2250#discussion_r532528452) [](ancestors = 532528452)

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**